### DISCUSSION TITLE
Problems with login on subpath

### DISCUSSION THREAD

**Discussion by ezeholz** [dani-garcia/vaultwarden#3096] [Q&A] [answered]
Hi all! I installed a few hours ago a clean instance of Vaultwarden in a local server I'm running. This server previously contained a few other services and all of them are forwarded to the Internet.
Since the first minutes of opening the web vault, it's stays permanently in "Loading" status. I later found out that, since I already used that server, vaultwarden it's trying to parse my localStorage, but it's unable to because some of the values are just strings and not objects as expect.

I bypassed that issue by creating the ones this app needs in order to see the login webpage by hand. After registering and playing a bit in the web app, I tried to connecting my phone app. First the web vault from the phone was having the same issue described above. But after a bit of tweaking on another service, I manage to get into the login webpage.
However, the Android app never recognice my Vaultwarden server, and just states that "We were unable to process your request. Please try again or contact us."

My server already has an active nginx instance, and a full Let's Encrypt certificate. In order to login in my server from outside, I managed to set a mTLS instance that let me consume every endpoint if I'm authenticated. In the other hand, if I'm logged in from my local network, the client certificate isn't necessary at all.

This is the config I set for Nginx:
```
server {
    listen 80 default_server;
    listen [::]:80;
    server_name xxx.com;

    location / {
      return 301 https://$server_name$request_uri;
    }
}

server {
  listen 4443 ssl;
  listen [::]:4443 ssl;

  server_name xxx.com;

  ssl_certificate      /etc/nginx/certificates/cert;
  ssl_certificate_key /etc/nginx/certificates/key;
  ssl_protocols       TLSv1.2 TLSv1.3;
  ssl_client_certificate /etc/nginx/certificates/CA;

  ssl_verify_depth 2;
  ssl_verify_client optional;

  set_real_ip_from 192.168.0.1;
  real_ip_header X-Forwarded-For;

  location / {
    if ($ssl_client_verify != "SUCCESS") { return 403; }
    proxy_pass https://localhost/;

    proxy_set_header Host $host;

    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header X-Forwarded-Host $host:$server_port;
    proxy_set_header X-Forwarded-Proto $scheme;
    proxy_set_header X-Forwarded-Server $host;
    proxy_set_header X-Real-IP $remote_addr;

    proxy_buffering off;

    proxy_buffer_size          128k;
    proxy_buffers              4 256k;
    proxy_busy_buffers_size    256k;

    proxy_http_version 1.1;
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection $http_connection;
  }

  location = /error.html {
    root /etc/nginx/html;
  }
}

server {
  listen 443 ssl;
  listen [::]:443 ssl;

  ssl_certificate      /etc/nginx/certificates/cert;
  ssl_certificate_key /etc/nginx/certificates/key;
  ssl_protocols       TLSv1.1 TLSv1.2 TLSv1.3;

  server_name xxx.com;

  proxy_buffering off;

  proxy_buffer_size          128k;
  proxy_buffers              4 256k;
  proxy_busy_buffers_size    256k;

  location /vault/ {
    satisfy any;

    deny 192.168.0.1;
    allow 192.168.0.0/24;
    deny all;
    auth_request /organizr-auth/1;

    proxy_pass http://127.0.0.1:3573;

    proxy_set_header Host $host;

    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header X-Forwarded-Proto $scheme;
    proxy_set_header X-Forwarded-Host $http_host;

    proxy_http_version 1.1;
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection $http_connection;
  }

  location ~ /organizr-auth/(.*) {
    internal;
    proxy_buffering on;
    proxy_pass http://127.0.0.1:8000/api/v2/auth/$1;
    proxy_set_header Content-Length "";
  }

  error_page 401 $scheme://$server_name/api/v2/organizr/error/$status?return=$request_uri;
  error_page 400 402 403 404 405 408 500 502 503 504  $scheme://$server_name/api/v2/organizr/error/$status;
}
```

And the following it's the docker compose I made for vaultwarden:
```
  vaultwarden:
    image: vaultwarden/server:latest
    container_name: vaultwarden
    restart: always
    environment:
      - 'WEBSOCKET_ENABLED=true'  # Enable WebSocket notifications.
      - 'DOMAIN=https://xxx.com/vault/'
      - 'DATABASE_URL=postgresql://xxx:xxx@postgres:5432/vaultwarden'
      - 'ADMIN_TOKEN=xxx-xxx'
    volumes:
      - ./config/bitwarden:/data
    ports:
      - 3573:80
      - 3012:3012
    restart: unless-stopped
    networks:
      - default
      - mailnet
      - drivenet
```

I can confirm that Vaultwarden it's running in 1.27.0, and I'll leave a screenshot of the data showed in the admin panel
![image](https://user-images.githubusercontent.com/30635862/210485721-c0895c1e-83bf-45ca-89fe-325e4e2db1ee.png)

Edit: As a way of heads up, I tried connecting from the extension app in Opera, and it works without any issues. I'm just asking to be able to connect to both of the devices I mostly use, and then I'll worry about making mTLS useful, or bypass it for this endpoint if it's way too complicate.

**Comment by BlackDex** [comment, marked as answer]
First the nginx config doesn't look complete or correct. Check the wiki regarding the proxy config and double check it https://github.com/dani-garcia/vaultwarden/wiki/Proxy-examples

Also, you have the `auth_request` module configured. This could cause some strange issues with the authentication headers maybe.

And further, check that no security headers are overruled or modified by nginx, like by ModSecurity for example. Those tend to break the workings of Vaultwarden. Vaultwarden already has all the needed security headers configured and sent when needed.

**Reply by ezeholz** [reply]
I updated the main discussion with the actual Nginx configuration, removing the parts that don't belong to vault itself.

As I said, I'm able to login in the web vault and in the extension app without any issues. And just when I try to log in from my Android Device it's when the message "We were unable to process your request. Please try again or contact us." appears.

**Reply by ezeholz** [reply]
The problem it's actually the same as a few others in issues reported, I'm missing a certificate to get the fullchain. Let me test it out and solve it, and that should fix an issue.
However, the issue of Vaultwarden trying to add it's own values to the localStorage remains, as it's unable to solve it on it's own

**Reply by BlackDex** [reply]
I have no issues with the local values.
Check the extensions of the browser you use, some _Privacy_ extensions block some stuff.

Also check your reverse proxy for headers it changes and blocking functionality as i mentioned before.

**Reply by BlackDex** [reply]
Check your ssl cert. If it is missing the cert chain it could cause issues. See https://github.com/dani-garcia/vaultwarden/wiki/Enabling-HTTPS#check-if-certificate-is-valid

**Reply by ezeholz** [reply]
Yep, that was it. After setting the fullchain into nginx, the Android app started working and I no longer have problems with the parse in localStorage. I don't really know why the later started to happen, but I tried again after the fullchain change and I wasn't able to reproduce it. Thanks!
